lib/jetstream/api/consumer.ex

```diff
@@ -1,47 +1,230 @@
 defmodule Jetstream.API.Consumer do
+  @moduledoc """
+  A module representing a NATS JetStream Consumer.
+
+  Learn more about consumers: https://docs.nats.io/nats-concepts/jetstream/consumers
+
+  ## The Jetstream.API.Consumer struct
+
+  The struct's only mandatory field to set is the `:stream_name`. The rest will have
+  the NATS default values set.
+
+  Note that consumers are ephemeral by default. Set the `:durable_name` to make it durable.
+
+  Consumer struct fields explanation:
+
+  * `:stream_name` - name of a stream the consumer is pointing at.
```

Comment:
I would be good to specify restrictions in this field. I know you can't use whitespace in stream_name et al.

Reply:
This field only references the stream. I've liknked [NATS docs](https://docs.nats.io/running-a-nats-service/nats_admin/jetstream_admin/naming) in the `Stream` struct's `:name` field.

Reply:
👍🏻